invenio_rdm_records/services/pids/providers/datacite.py

```diff
@@ -24,6 +25,25 @@ def __init__(self, username, password, prefix, test_mode, **kwards):
         self.test_mode = test_mode
 
 
+class DataCiteClientFromConfig(DataCiteClient):
```

Comment:
I would simply use this class as `DataCiteClient` since I think 100% of the cases args will come from config.